### PR TITLE
Updating the GitHub Action File

### PR DESCRIPTION
**Pull request by saketjajoo**
This PR updates the GH Actions file to have the following changes:
- Renaming the steps in the workflow to make it more meaningful (code readability).
- Disabling prombench trigger on release cut (prombench now works only upon commenting on a PR).
- Adding comments to a PR when a prombench run is cancelled.
- Adding labels to a PR when a prombench run is triggered.
- Fixing the docker image URLs to point to the GHCR repo instead of a personal dockerhub account.

### CODE REVIEW

File: .github/workflows/prombench-gmp-collector.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 name: "Run Prombench"
 
 on:
-  push:
-    tags:
-      - 'v*-gmp.*'
+  # push:
```
Comment (maxamins) [inline .github/workflows/prombench-gmp-collector.yml]: Delete this. If needed we can figure out how to do it.

File: .github/workflows/prombench-gmp-collector.yml [hunk 2]
```diff
@@ -36,9 +36,11 @@ jobs:
     runs-on: ubuntu-latest
     permissions:
       pull-requests: write
+      contents: read
```
Comment (maxamins) [inline .github/workflows/prombench-gmp-collector.yml]: Why do you need these permissions?

File: .github/workflows/prombench-gmp-collector.yml [hunk 3]
```diff
@@ -70,7 +72,7 @@ jobs:
         uses: 'google-github-actions/setup-gcloud@v1'
 
       
-      - name: check_benchmark_trigger_action
+      - name: 'Set environment variables based on the triggered action'
         run: |
           GMP_COLLECTOR_TAG=$(echo ${{ github.event.comment.body }} | head -n 1 | awk '{ print $2 }')
```
Comment (maxamins) [inline .github/workflows/prombench-gmp-collector.yml]: This is quite confusing. How about `COLLECTOR_TAG_OR_CANCEL`

File: .github/workflows/prombench-gmp-collector.yml [hunk 4]
```diff
@@ -126,9 +128,18 @@ jobs:
...
+          
+      - name: 'Adding the prombench label to the PR'
+        if: env.CANCEL_BENCHMARK == 'false'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          curl --silent -L -X POST -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GITHUB_TOKEN" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/${GITHUB_ORG}/${GITHUB_REPO}/issues/${PR_NUMBER}/labels \
+            -d '{"labels":["prombench"]}' || true
```
Comment (maxamins) [inline .github/workflows/prombench-gmp-collector.yml]: what is `|| true` doing?

**Review by maxamins — COMMENTED** [review]
Good job!

File: .github/workflows/prombench-gmp-collector.yml (hunk 2 above)
Comment (saketjajoo) [reply .github/workflows/prombench-gmp-collector.yml]: This is done to only have the necessary permissions for the `GITHUB_TOKEN`. The [default permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) are not required as the token is only used for commenting on a PR and reading docker images.
Ref: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token

File: .github/workflows/prombench-gmp-collector.yml (hunk 4 above)
Comment (saketjajoo) [reply .github/workflows/prombench-gmp-collector.yml]: It's just to ensure that even if the command fails for some reason, the workflow does not stop. If the add label command fails, the exit code is non zero which will stop the remaining steps in the workflow. The `|| true ` ensures that the exit code is always `0`.

Comment (maxamins) [reply .github/workflows/prombench-gmp-collector.yml]: Wouldn't you need it for the comment in line 122.

Comment (maxamins) [reply .github/workflows/prombench-gmp-collector.yml]: Let's add failure resistance when we migrate from curl to gh

File: .github/workflows/prombench-gmp-collector.yml (hunk 2 above)
Comment (maxamins) [reply .github/workflows/prombench-gmp-collector.yml]: Let's add this section if end up having permission issues.

**Review by maxamins — APPROVED** [review]

**Comment by maxamins** [comment]
Thanks!

**Pull request merged**